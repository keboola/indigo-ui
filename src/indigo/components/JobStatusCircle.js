import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircle } from '@fortawesome/free-solid-svg-icons';

const statusColorMap = {
  success: 'success',
  error: 'danger',
  warn: 'danger',
  warning: 'danger',
  processing: 'warning',
  waiting: 'default',
  cancelled: 'canceled',
  terminating: 'canceled',
  terminated: 'canceled',
};

class JobStatusCircle extends React.Component {
  render() {
    const statusColor = statusColorMap[this.props.status] || 'default';
    return <FontAwesomeIcon icon={faCircle} className={`job-status-circle-${statusColor}`} />;
  }
}

JobStatusCircle.propTypes = {
  status: PropTypes.string,
};

export default JobStatusCircle;
