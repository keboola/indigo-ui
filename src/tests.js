import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
