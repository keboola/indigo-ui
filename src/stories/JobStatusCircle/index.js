import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';

import JobStatusCircle from '../../indigo/components/JobStatusCircle';

storiesOf('JobStatusCircle', module).add(
  'Job Status - Default',
  withInfo({
    inline: true,
  })(() => {
    return <JobStatusCircle status="default"/>;
  })
).add(
  'Job Status - Success',
  withInfo({
    inline: true,
  })(() => {
    return <JobStatusCircle status="success"/>;
  })
).add(
  'Job Status - Danger',
  withInfo({
    inline: true,
  })(() => {
    return <JobStatusCircle status="error"/>;
  })
).add(
  'Job Status - Warning',
  withInfo({
    inline: true,
  })(() => {
    return <JobStatusCircle status="processing"/>;
  })
);
