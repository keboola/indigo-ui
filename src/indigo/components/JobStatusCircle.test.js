import React from 'react';

import { matchesSnapshot } from '../../tests';
import JobStatusCircle from './JobStatusCircle';

describe('<JobStatusCircle />', () => {
  it('Job Status - Cancelled', () => {
    matchesSnapshot(<JobStatusCircle status="cancelled" />);
  });

  it('Job Status - Success', () => {
    matchesSnapshot(<JobStatusCircle status="success" />);
  });

  it('Job Status - Error', () => {
    matchesSnapshot(<JobStatusCircle status="error" />);
  });

  it('Job Status - Warning', () => {
    matchesSnapshot(<JobStatusCircle status="processing" />);
  });

  it('Job Status - Terminated', () => {
    matchesSnapshot(<JobStatusCircle status="terminated" />);
  });

  it('Job Status - not provided', () => {
    matchesSnapshot(<JobStatusCircle />);
  });

  it('Job Status - random status', () => {
    matchesSnapshot(<JobStatusCircle status="random-status" />);
  });
});
