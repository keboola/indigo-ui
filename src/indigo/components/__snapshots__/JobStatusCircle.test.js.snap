// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobStatusCircle /> Job Status - Cancelled 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-canceled"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - Error 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-danger"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - Success 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-success"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - Terminated 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-canceled"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - Warning 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-warning"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - not provided 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-default"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<JobStatusCircle /> Job Status - random status 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 job-status-circle-default"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
